### PR TITLE
Split C std library into declarations and definitions

### PR DESCRIPTION
**Pull request by XuJiandong**
The header files are located in "libc/include" folder.
The src files are located in "libc/src" folder.

### CODE REVIEW

**Comment by XuJiandong** [comment]
request to review. @xxuejie

**Comment by XuJiandong** [comment]
When this pull request is accepted, there will be another change on this: https://github.com/nervosnetwork/ckb-miscellaneous-scripts/pull/3. (modified according to the new interface).

File: libc/memory.h [hunk 1]
```diff
@@ -1,2 +1,3 @@
+
```
Comment (xxuejie) [inline libc/memory.h]: Maybe remove this line?

File: libc/src/impl.c [hunk 2]
```diff
@@ -0,0 +1,1193 @@
...
+// and then done. It automatically includes impl.c(this file). It is compatible
+// to a lot of old CKB code.
+//
+// If you have more than one C file, and it's a big project. You need to define
+// macro CKB_DECLARATION_ONLY for every C files. Compile very C file, together
+// with this impl.c (compiled *without* macro CKB_DECLARATION_ONLY) and then
+// link them together.
+//
+
+#ifndef __IMPL_INCLUDED__
```
Comment (xxuejie) [inline libc/src/impl.c]: Let's add a `CKB_` prefix to make it `__CKB_IMPL_INCLUDED__`, so as to avoid unnecessary conflicts.

**Review by xxuejie — APPROVED** [review]
Just some minor comments, then we can merge this.

**Pull request merged**